fuel/streams.py

```diff
@@ -135,3 +138,42 @@ def get_epoch_iterator(self, **kwargs):
         else:
             self._fresh_state = False
         return super(DataStream, self).get_epoch_iterator(**kwargs)
+
+
+class ServerDataStream(AbstractDataStream):
```

Comment:
(I think ZMQ has some sort of side-channel things that IPython uses for this kind of riff-raff metadata.)